circle.yml

```diff
@@ -0,0 +1,135 @@
+dependencies:
+  cache_directories:
+    - ~/.android
+    - ~/.gradle
+  override:
+    - echo "dummy"
+  pre:
+    - sudo apt-get update > /dev/null 2> /dev/null
+    - sudo apt-get install xvfb > /dev/null 2> /dev/null
+    - sudo apt-get install xdotool > /dev/null 2> /dev/null
+    - sudo apt-get install telnet > /dev/null 2> /dev/null
+    - sudo apt-get install x11-utils > /dev/null 2> /dev/null
+    - sudo apt-get install xvkbd > /dev/null 2> /dev/null
+
+    - gradle -v
+    - cd ~ ; wget https://services.gradle.org/distributions/gradle-2.10-bin.zip
+    - mkdir ~/g ; cd ~/g ; unzip ~/gradle-2.10-bin.zip
+    - gradle -v
+
+    - echo y | android update sdk --no-ui --all --filter "tools"
+    #- android list sdk --all | grep -i tools
+    #- android list sdk --all --extended
+
+    - echo y | android update sdk --no-ui --all --filter build-tools-23.0.3 > /dev/null 2> /dev/null
+    - echo y | android update sdk --no-ui --all --filter build-tools-23.0.1 > /dev/null 2> /dev/null
+    - echo y | android update sdk --no-ui --all --filter android-23 > /dev/null 2> /dev/null
+    - echo y | android update sdk --no-ui --all --filter platform-tools > /dev/null 2> /dev/null
+
+    - echo y | android update sdk --no-ui --all --filter extra-android-m2repository > /dev/null 2> /dev/null
+    - echo y | android update sdk --no-ui --all --filter extra-google-m2repository > /dev/null 2> /dev/null
+    - echo y | android update sdk --no-ui --all --filter extra-android-support > /dev/null 2> /dev/null
+
+    - echo y | android update sdk --no-ui --all --filter platform-tools-preview > /dev/null 2> /dev/null
+    - echo y | android update sdk --no-ui --all --filter sys-img-armeabi-v7a-android-23 > /dev/null 2> /dev/null
+    - echo y | android update sdk --no-ui --all --filter sys-img-armeabi-v7a-addon-google_apis-google-23 > /dev/null 2> /dev/null
+    - echo y | android update sdk --no-ui --all --filter addon-google_apis-google-23 > /dev/null 2> /dev/null
+
+    - pwd ; ls -al
+
+    - git submodule init
+    - git submodule update
+    - gradle build --stacktrace --info
+    - find . -name '*.apk' ; exit 0
+
+    - mkdir -p app/build/outputs/apk/
+    - cp -v ./build/outputs/apk/Etar-Calendar-debug-unaligned.apk app/build/outputs/apk/app-debug-unaligned.apk
+
+    - zip -d app/build/outputs/apk/app-debug-unaligned.apk META-INF/\*     # remove signature !!
+    - cp -av app/build/outputs/apk/app-debug-unaligned.apk ~/app.apk ;
+      cd ~/ ;
+      echo xxxxxxrm -f ~/.android/debug.keystore ;
+      ls -al ~/.android/debug.keystore ;
+      if [ ! -f ~/.android/debug.keystore ]; then echo "*** generating new signer key ***" ;
+      echo "*** generating new signer key ***";
+      echo "*** generating new signer key ***";
+      keytool -genkey -v -keystore ~/.android/debug.keystore -storepass android -keyalg RSA -keysize 2048 -validity 10000 -alias androiddebugkey -keypass android -dname "CN=Android Debug,O=Android,C=US" ;
```

Comment:
it does NOT !
circle CI will save it between runs. a new key will only be created on the first run.
hence the "if [ ! -f ~/.android/debug.keystore ]"

echo xxxxxxrm -f ~/.android/debug.keystore ; --> this line is only there to quickly make a new key if needed.

but its your call. i run many apps with this logic :-)